src/cascadia/PublicTerminalCore/HwndTerminal.cpp

```diff
@@ -139,27 +171,42 @@ void HwndTerminal::RegisterScrollCallback(std::function<void(int, int, int)> cal
     _terminal->SetScrollPositionChangedCallback(callback);
 }
 
-void HwndTerminal::RegisterWriteCallback(const void _stdcall callback(wchar_t*))
+void HwndTerminal::_WriteTextToConnection(const std::wstring& input) noexcept

```

Comment:
Opportunity for significant clarity-enhancing.

```cpp
void HwndTerminal::_WriteTextToConnection(const std::wstring& input) noexcept
{
    if (!_pfnWriteCallback)
    {
        return;
    }

    try
    {
        auto callingText{ wil::make_cotaskmem_string(input.data(), input.size()) };
        _pfnWriteCallback(callingText.release());
    }
    CATCH_LOG();
}
```

we release the string if we call the callback with it (so it becomes owned by the CLR, if i remember?) and we use the wil helpers to make it all smooth and not use wcscpy.

Reply:
That is significantly clearer, thanks!

Reply:
i certainly hope it _works_ still 😁